examples/nixnet/lin-signal-single-point-output.py

```diff
@@ -0,0 +1,149 @@
+
+For instructions on how to use protoc to generate gRPC client interfaces, see our "Creating a gRPC
+Client" wiki page:
+  https://github.com/ni/grpc-device/wiki/Creating-a-gRPC-Client
+
+Refer to the NI XNET gRPC Wiki for the latest C Function Reference:
+  https://github.com/ni/grpc-device/wiki/NI-XNET-C-Function-Reference
+ 
+ Running from command line:
+Server machine's IP address, port number, and interface name can be passed as separate command line
+arguments.
+  > python lin-signal-single-point-output.py <server_address> <port_number> <interface_name>
+If they are not passed in as command line arguments, then by default the server address will be
+"localhost:31763"
+"""
+
+import sys
+
+import grpc
+import nixnet_pb2 as nixnet_types
+import nixnet_pb2_grpc as grpc_nixnet
+
+SERVER_ADDRESS = "localhost"
+SERVER_PORT = "31763"
+INTERFACE = "LIN1"
+
+# Read in cmd args
+if len(sys.argv) >= 2:
+    SERVER_ADDRESS = sys.argv[1]
+if len(sys.argv) >= 3:
+    SERVER_PORT = sys.argv[2]
+if len(sys.argv) >= 4:
+    INTERFACE = sys.argv[3]
+
+# Parameters
+DATABASE = "NIXNET_exampleLDF"
+CLUSTER = "Cluster"
+SIGNAL_LIST = "MasterSignal1_U16,MasterSignal2_U16"
+NUM_SIGNALS = 2
+
+
+def check_for_error(status):
+    """Raise an exception if the status indicates an error."""
+    if status != 0:
+        error_message_response = client.StatusToString(
+            nixnet_types.StatusToStringRequest(status_id=status)
+        )
+        raise Exception(error_message_response.status_description)
+
+
+i = 0
+value_buffer = [0.0] * NUM_SIGNALS
+
+# Create the communication channel for the remote host and create connections to the NI-XNET and
+# session services.
+channel = grpc.insecure_channel(f"{SERVER_ADDRESS}:{SERVER_PORT}")
+client = grpc_nixnet.NiXnetStub(channel)
+
+# Change this to set the interface to slave mode
+IsMaster = 1
+
+# The schedule is identified by its index. The index is mapped to
+# the schedules in the database. Index 0 is the first schedule
+# displayed in the Database Editor.
+ScheduleIndex = 0
+
+# Display parameters that will be used for the example.
+print("Interface: " + INTERFACE, "Database: " + DATABASE, "Signal List: " + SIGNAL_LIST, sep="\n")
+
+try:
+    if IsMaster != 0:
+        print("Master?: Yes\n")
+    else:
+        print("Master?: No\n")
+
+    # Create an XNET session in SignalOutSinglePoint mode
+    create_session_response = client.CreateSession(
+        nixnet_types.CreateSessionRequest(
+            database_name=DATABASE,
+            cluster_name=CLUSTER,
+            list=SIGNAL_LIST,
+            interface=INTERFACE,
+            mode=nixnet_types.CREATE_SESSION_MODE_MODE_SIGNAL_OUT_SINGLE_POINT,
+        )
+    )
+    check_for_error(create_session_response.status)
+
+    session = create_session_response.session_ref
+    print("Session Created Successfully. \n")
+
+    if IsMaster != 0:
+        # Set the schedule - this will also automatically enable master mode
```

Comment:
We discussed this with Waldemar to understand details of LIN. In a network there can only be one master, and rest of the interfaces should be run as slaves. And as you mentioned, master will be responsible for setting the schedule so that slaves can write frames based on the schedule. If there is no interface configured as master, there would be no output.

The mode is also device (interface) specific, not session specific. So you can't have two sessions for same interface, and one of them trying to configure interface as master and other session trying to configure it as slave.

We have ported this example as is from C examples that XNET ships. What is intended from user here is, they will run one instance of example with `IsMaster=1` to configure the master and one or more instances with `IsMaster=0` and also pointing to different interfaces as well. Users of LIN should be aware of this setup required for LIN. So I think explicit comment is not required for this.

Another interesting thing that came up during discussion was, in a network if we try to configure two instances as Master and the API won't actually return any error. But since this is invalid setup we will see error during communication later as two interfaces will try to control the schedule.